templates/gRPC-C++.podspec.template

```diff
@@ -20,108 +20,100 @@
+    for lib_name in all_grpc_libs_and_abseil_specs(lib)[0]:
+      lib = lib_maps[lib_name]
+      for field in fields:
+        files.update(lib.get(field, []))
+    return list(sorted(files))
+
+  def grpc_filegroup_files(expect_filegroups, groups):
     out = []
-    for lib in libs:
-      if lib.name in expect_libs:
-        for group in groups:
-          out += lib.get(group, [])
-    return sorted(out)
+    for filegroup_name in expect_filegroups:
+      filegroup = filegroup_maps[filegroup_name]
+      for group in groups:
+        out += filegroup.get(group, [])
+    return list(sorted(set(out)))
 
   def filter_grpcpp(files):
-    return [file for file in files if not file.startswith("include/grpc++")]
-
-  grpc_libs = ("grpc", "gpr", "upb")
-
-  def grpc_private_files(libs):
-    out = grpc_lib_files(libs, grpc_libs, ("headers", "src"))
-    return out
-
-  def grpc_private_headers(libs):
-    out = grpc_lib_files(libs, grpc_libs, ("headers",))
-    return out
-
-  def grpc_public_headers(libs):
-    out = grpc_lib_files(libs, grpc_libs, ("public_headers",))
-    return out
-
-  def grpcpp_proto_files(filegroups):
-    out = grpc_lib_files(filegroups, ("grpc++_codegen_proto", "grpc++_config_proto"), ("headers", "src", "public_headers"))
-    excl_files = grpc_lib_files(filegroups, ("grpc++_codegen_base",), ("headers", "src", "public_headers"))
-    out = [file for file in out if file not in excl_files]
-    out = filter_grpcpp(out)
-    return out
-
-  def grpcpp_private_files(libs, filegroups):
-    out = grpc_lib_files(libs, ("grpc++",), ("headers", "src"))
-    excl_files = grpc_private_files(libs)
-
+    return sorted([file for file in files if not file.startswith("include/grpc++")])
+
+  grpc_files = grpc_all_lib_files("grpc", ("public_headers", "headers", "src"))
+  grpc_private_files = grpc_all_lib_files("grpc", ("headers", "src"))
+  grpc_public_headers = grpc_all_lib_files("grpc", ("public_headers",))
+  grpc_private_headers = grpc_all_lib_files("grpc", ("headers",))
+
+  grpcpp_proto_files = filter_grpcpp(
+    set(grpc_filegroup_files(("grpc++_codegen_proto", "grpc++_config_proto"), ("headers", "src", "public_headers")))
+    - set(grpc_filegroup_files(("grpc++_codegen_base",), ("headers", "src", "public_headers")))
+  )
+  grpcpp_private_files = filter_grpcpp(
+    set(grpc_all_lib_files("grpc++", ("headers", "src")))
+    - set(grpc_files)
     # We exclude proto related files in this particular podspec so that we can provide a protobuf-independent lib
-    excl_files += grpcpp_proto_files(filegroups)
-    out = [file for file in out if file not in excl_files]
-
-    out = filter_grpcpp(out)
-    return out
-
-  def grpcpp_private_headers(libs, filegroups):
-    out = grpc_lib_files(libs, ("grpc++",), ("headers",))
-    excl_files = grpc_private_headers(libs)
-
+    - set(grpcpp_proto_files)
+  )
+  grpcpp_private_headers = filter_grpcpp(
+    set(grpc_all_lib_files("grpc++", ("headers",)))
+    - set(grpc_files)
```

Comment:
`grpc_private_headers` to keep consistency with the old version's behavior?

Reply:
Done.